### PR TITLE
Corrected the help for exportbundle command

### PR DESCRIPTION
**Pull request by vinu2003**
This PR has code changes that fix the help for export bundle command. 
There shouldn't be --output and --format options in thge help menu. 

## QA steps
execute juju export-bundle --help command.

### CODE REVIEW

File: cmd/juju/model/exportbundle.go [hunk 1]
```diff
@@ -55,7 +54,7 @@ func (c *exportBundleCommand) Info() *cmd.Info {
 // SetFlags implements Command.
 func (c *exportBundleCommand) SetFlags(f *gnuflag.FlagSet) {
 	c.ModelCommandBase.SetFlags(f)
-	c.out.AddFlags(f, "yaml", output.DefaultFormatters)
+	//c.out.AddFlags(f, "yaml", output.DefaultFormatters)
```
Comment (anastasiamac) [inline cmd/juju/model/exportbundle.go]: Delete this all together if not needed?...

Comment (vinu2003) [reply cmd/juju/model/exportbundle.go]: we dont need to have --format and --output options. AddFLags just injects both cli flags. So i removed.

Comment (howbazaar) [reply cmd/juju/model/exportbundle.go]: You'll need to modify the body of the run method too most likely...

Comment (anastasiamac) [reply cmd/juju/model/exportbundle.go]: My original comment was to point out that it is not good enough to comment this code out. Remove it! 
We don not want to clatter our production code with commented out code :D

**Comment by vinu2003** [comment]
Hi @howbazaar ,

I havent read your comment. But i just fixed the stdout print as below,
c.out.WriteFormatter(ctx, cmd.FormatYaml, result)

i didnt use fmt.Fprintf()...

question:
Why have you suggested ioutil.writeFile ?

Also,
You have mentioned let user decide the extension and we should not append it to filename.
I have done that and pushed another commit.

But, the GUI appends as '<modelname>.yaml'.
Please let me know your input i can revert back that if we should match with GUI.

**Comment by mitechie** [comment]
For the GUI comparison, the GUI bdoesnt allow you to provide a requested
filename. You just get the same name each time. Since, in the CLI case, we
allow the user to express a filename opinion infeel we should respect it.
If they script some action and provide a filename I believe they'd expect
to be able to use the same filename later in their scripting without issue.

**Comment by vinu2003** [comment]
back and just verified with fmt.Printf() the pipe issie disappeared. So i beleive all cosmetic issues are fixed for bundle export.

**Comment by vinu2003** [comment]
Hi @mitechie Agree with your point.
Thanks for pointing this out. I have also verified it is not mandatory to be with .yaml extension to get deployed. So its all ok for now.

**Comment by mitechie** [comment]
QA ok on the changes made. The help is improved and the format of the terminal output is reusable when piped to a file correctly.

Can you also edit the Summary please? 

    Exports the current model configuration as bundle.

to 

    Exports the current model configuration as a reusable bundle.

Also please add a newline after the example description to line up with the help output template
https://docs.google.com/document/d/1ySjCNqd0x6veLfcBetxLI9NH7qfw3xayLWJNqXMvyW8/edit

In that example can you add the .yaml to the example filename so that it shows that it's respecting what the user types and encourages them to have that even though we don't require it?

Thanks for the updates!

**Comment by vinu2003** [comment]
Hi @mitechie ,
Thanks again for your reviews. I have made the changes requested.

File: api/bundle/client.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func NewClient(st base.APICallCloser) *Client {
 func (c *Client) ExportBundle() (string, error) {
 	var result params.StringResult
 	if bestVer := c.BestAPIVersion(); bestVer < 2 {
-		return "", errors.Errorf("ExportBundle() on v%d is not supported", bestVer)
+		return "", errors.Errorf("this controller version v%d doen not support bundle feature.", bestVer)
```
Comment (mitechie) [inline api/bundle/client.go]: This output a v1 in the QA work. That's not the controller version so this turns into a bit of misinformation doesn't it?

File: api/bundle/client_test.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ func (s *bundleMockSuite) TestFailExportBundlev1(c *gc.C) {
 	)
 	result, err := client.ExportBundle()
 	c.Assert(err, gc.NotNil)
-	c.Assert(err.Error(), gc.Equals, "ExportBundle() on v1 is not supported")
+	c.Assert(err.Error(), gc.Equals, "this controller version v1 doen not support bundle feature.")
```
Comment (mitechie) [inline api/bundle/client_test.go]: same here, v1 isn't the controller version

File: cmd/juju/model/exportbundle.go [hunk 4]
```diff
@@ -32,30 +31,30 @@ type exportBundleCommand struct {
 }
 
 const exportBundleHelpDoc = `
-Exports the current model configuration as bundle.
+Exports the current model configuration as a reusable bundle.
```
Comment (mitechie) [inline cmd/juju/model/exportbundle.go]: thank you

File: cmd/juju/model/exportbundle.go [hunk 5]
```diff
@@ -32,30 +31,30 @@ type exportBundleCommand struct {
...
 
 If --filename is not used, the configuration is printed to stdout.
  --filename specifies an output file.
 
 Examples:
 
     juju export-bundle
-	juju export-bundle --filename mymodel
+	juju export-bundle --filename mymodel.yaml
```
Comment (mitechie) [inline cmd/juju/model/exportbundle.go]: <3

**Comment by vinu2003** [comment]
$$merge$$

**Pull request merged**